src/Catalyst.Core.Lib/Network/DnsClient.cs

```diff
@@ -77,17 +77,20 @@ public async Task<IDnsQueryResponse> GetTxtRecordsAsync(string hostname)
         public IEnumerable<PeerId> GetSeedNodesFromDns(IEnumerable<string> seedServers)
         {
             var peers = new List<PeerId>();
-            seedServers.ToList().ForEach(async seedServer =>
+
+            async void Action(string seedServer)
```

Comment:
async voids are against the Async pattern
https://msdn.microsoft.com/en-us/magazine/jj991977.aspx

> Async void methods have different error-handling semantics. Exceptions escaping from PrimeCustomTask will be very awkward to handle.
Async void methods have different composing semantics. This is an argument centered around code maintainability and reuse. Essentially, the logic in PrimeCustomTask is there and that's it - it can't be composed into a higher-level async method.
Async void methods are difficult to test. Following naturally from the first two points, it's very difficult to write a unit test covering PrimeCustomTask (or anything that calls it).